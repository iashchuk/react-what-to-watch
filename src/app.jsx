import React, { createContext } from "react";

import MainPage from "./pages/main-page/main-page";
import DetailsPage from "./pages/details-page/details-page";

import { data } from "../src/api/mocks";


export const Context = createContext();

export const options = {
  data
};

const getPage = () => {
  switch (location.pathname) {
    case `/details`:
      return <DetailsPage />;
    default:
      return <MainPage />;
  }
};

const App = () => {
  return <Context.Provider value={options}>{getPage()}</Context.Provider>;
};

export default App;
