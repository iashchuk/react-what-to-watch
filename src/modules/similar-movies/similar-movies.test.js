import React from "react";
import renderer from "react-test-renderer";
import { Context, options } from "../../app";
import SimilarMovies from "./similar-movies.jsx";

describe(`SimilarMovies`, () => {
  it(`renders correctly`, () => {
    const dom = (
      <Context.Provider value={options}>
        <SimilarMovies />
      </Context.Provider>
    );
    const component = renderer.create(dom);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
