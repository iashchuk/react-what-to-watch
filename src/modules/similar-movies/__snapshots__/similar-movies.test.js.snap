// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimilarMovies renders correctly 1`] = `
<section
  className="catalog catalog--like-this"
>
  <h2
    className="catalog__title"
  >
    More like this
  </h2>
  <div
    className="catalog__movies-list"
  >
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="Fantastic Beasts: The Crimes of Grindelwald"
            height="175"
            src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="Bohemian Rhapsody"
            height="175"
            src="img/bohemian-rhapsody.jpg"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
          >
            Bohemian Rhapsody
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="Macbeth"
            height="175"
            src="img/macbeth.jpg"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
          >
            Macbeth
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <img
            alt="Aviator"
            height="175"
            src="img/aviator.jpg"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
          >
            Aviator
          </a>
        </h3>
      </article>
    </div>
  </div>
</section>
`;
